File: 1476-count-negative-numbers-in-a-sorted-matrix/count-negative-numbers-in-a-sorted-matrix.go
func countNegatives(grid [][]int) int {
    rows := len(grid)
    cols := len(grid[0])
    rowIdx := 0
    colIdx := cols - 1
    ans := 0
    for ;rowIdx < rows && colIdx >=0; {
        if(grid[rowIdx][colIdx] < 0){
            ans += (rows - rowIdx)
            colIdx--
        }else{
            rowIdx++
        }
    }
    return ans
}

File: 1476-count-negative-numbers-in-a-sorted-matrix/count-negative-numbers-in-a-sorted-matrix.cpp
class Solution {
public:
    int countNegatives(vector<vector<int>>& grid) {
        int rows = grid.size(), cols = grid[0].size();
        int rowIdx = 0, colIdx = cols - 1;
        int ans = 0;
        while(rowIdx < rows && colIdx >= 0){
            if(grid[rowIdx][colIdx] < 0){
                ans += (rows - rowIdx);
                colIdx--;
            }else{
                rowIdx++;
            }
        }
        return ans;
    }
};